### PR TITLE
♿  amp-lightbox: Improve accessibility with focus

### LINKED ISSUE #27114: [amp-lightbox] focus outside should close the lightbox
## Describe the new feature or change to an existing feature you'd like to see

When I use a **Screen reader**, since the focus is not trapped in the lightbox I can tab and continue my reading in the page but the lightbox stays open. We known it's possible to close with esc but will the user know/do it ? I think a focus outside should be considered as a click outside... so closing it.

## Additional irritant

We had to add a close event that focus on the trigger that opened the lightbox so if we close the lightbox later in the read, then the focus go back to this button and I loose where I am at.

When 2 buttons open the same lightbox, I didn't find a way to go back to the good trigger so I had to duplicate the lightbox and that shouldn't be required

### PR DESCRIPTION
**Pull request by catherinehoule**
- Implement focussing on the lightbox on open by :
    - Validates if there already is a focus in it, if not;
    - Look for a close button and focus on it if there is, then if not;
    - Creates a SR close button and focus on it
- Implement the return of focus on the trigger element when lightbox closes
- Fix the i-amphtml-ad-close-header that was closing only once and didn't close on enter

Note: Adapted getExistingCloseButton_ and createScreenReaderCloseButton_ from sidebar

Fixes #27114

### CODE REVIEW

File: extensions/amp-lightbox/amp-lightbox.md [hunk 1]
```diff
@@ -164,6 +169,8 @@ In the examples below, we demonstrate how the transition looks for an AMPHTML ad
 </noscript>
 </amp-img>
 
+[/filter]<!-- formats="ads" -->
```
Comment (caroqliu) [inline extensions/amp-lightbox/amp-lightbox.md:172]: Why is this line moved?

File: extensions/amp-lightbox/amp-lightbox.md [hunk 2]
```diff
@@ -78,6 +80,11 @@ Must be set to `nodisplay`.
 
 [filter formats="websites, stories"]
 
+#### `data-close-button-aria-label` (optional)
+
+Optional attribute used to set ARIA label for the close button added for accessibility.
```
Comment (caroqliu) [inline extensions/amp-lightbox/amp-lightbox.md:85]:
```suggestion
Optional attribute used to set ARIA label for the autogenerated close button added for accessibility. By default the applied label is "Close the modal."
```

File: extensions/amp-lightbox/amp-lightbox.md [hunk 3]
```diff
@@ -30,7 +30,9 @@ limitations under the License.
 
 The `amp-lightbox` component defines child elements that display in a full-viewport overlay/modal. When the user taps or clicks an element (e.g., a button), the `amp-lightbox` ID referenced in the clicked element's `on` attribute triggers the lightbox to take up the full viewport and displays the child elements of the `amp-lightbox`.
 
-Pressing the escape key on the keyboard closes the lightbox. Alternatively, setting the `on` attribute on one or more elements within the lightbox and setting its method to `close` closes the lightbox when the element is tapped or clicked.
+Pressing the escape key on the keyboard or setting focus on an element outside the lightbox closes the lightbox. Alternatively, setting the `on` attribute on one or more elements within the lightbox and setting its method to `close` closes the lightbox when the element is tapped or clicked. Once closed, the focus will be sent back to the trigger element.
```
Comment (caroqliu) [inline extensions/amp-lightbox/amp-lightbox.md:33]: Let's put this in a `### Accessibility` header.

File: extensions/amp-lightbox/amp-lightbox.md (hunk 1 above)
Comment (catherinehoule) [reply extensions/amp-lightbox/amp-lightbox.md:172]: When I looked at the doc here, at first I got confused https://amp.dev/documentation/components/amp-lightbox/?format=websites The doc that's in between is talking about ads features. It seemed like it should be in the ad filter, specially that the title of the ad section is in the ad filter. So you have actions, then the big title says transparent background are disallowed and in small it's about the ad. But I can put it back where it was if you think it should stay in the web part.

**Comment by caroqliu** [comment]
Looks like you should merge or rebase with `master` to get the Travis tests to run.

**Comment by catherinehoule** [comment]
@caroqliu @powerivq I rebased earlier and everything passed. Would one of you merge it ?
